File: front-end/src/application/usecase/workflow/workflow-import-use-case.ts
import { Either } from '@ull-tfg/ull-tfg-typescript';
import type { DataError } from '@ull-tfg/ull-tfg-typescript';
import { ExportFile } from '@/domain/value-object/export-file';
import { WorkFlow } from '@/domain/entity/workflow';

export interface WorkFlowImportUseCase {
  /**
   * Import a workflow from an ExportFile.
   * 
   * @param file - The ExportFile to import
   * @returns The imported workflow
   */
  importWorkFlow(file: ExportFile): Promise<Either<DataError, WorkFlow>>;
}

File: back-end/src/main/java/es/ull/project/application/usecase/workflow/WorkFlowImportUseCase.java
package es.ull.project.application.usecase.workflow;

import java.util.Optional;

import es.ull.project.domain.entity.WorkFlow;
import es.ull.project.domain.valueobject.ExportFile;

public interface WorkFlowImportUseCase {
    /**
     * Import a workflow entity from a JSON file and return the workflow object.
     *
     * @param exportFile the ExportFile containing the workflow data
     * @return an Optional containing the workflow if import is successful, or empty if it fails
     */
    Optional<WorkFlow> importWorkFlow(ExportFile exportFile);
}
